LeetcodeProblemsTests/Algorithms/2Sum_Test.js

```diff
@@ -2,18 +2,16 @@ const assert = require("assert");
 const twoSum = require("../../LeetcodeProblems/Algorithms/2Sum").twoSum;
 const twoSum2 = require("../../LeetcodeProblems/Algorithms/2Sum").twoSum2;
 
+
 var test = function () {
   assert.deepEqual([0,1], twoSum([2,7,11,15], 9));
   assert.deepEqual([1,2], twoSum([3,2,4], 6));
   assert.deepEqual([0,1], twoSum([3,3], 6));
-};
-
-var test2 = function () {
```

Comment:
Can we leave two functions for the two different solutions? That way both cases are isolated.

Reply:
Not really. Infact, the way the file was earlier with two functions, the second function 'test2' was actually not running, so the other solution wasn't even being tested. You can add some console.log statements to confirm the same but you can also have a look directly at the code that corresponds to running the test functions which can be found [here](https://github.com/ignacio-chiazzo/Algorithms-Leetcode-Javascript/blob/master/Test.js#L16). It can be seen that only the 'test' method gets invoked, if we required 'test2' to be invoked we'd have to add that in there.

So what do we do if we have 10 solutions to a problem? I don't think adding 'test2', 'test3', ...'test10' is a good way to do it, besides having to deal with the exception handling for files that do not have the 'testX' method.

So maybe having just one 'test' method and testing every solution to the problem within that method was what came to my mind. Let me know if this reasoning doesn't make sense.

Reply:
I personally like to have one test function for a solution so that if there is a bug in a particular solution the specific test fails.

Reply:
Well ideally I would like that to happen too but I'm not sure how we can achieve that in a clean manner without a lot of hardcoding.

Essentially, we would want to have multiple test functions exported per test file, and somehow loop through all of the functions in the import without knowing the name explicitly.

I haven't coded a lot in JavaScript so I'll need a bit of time to check. I think it's possible to do it though, as the import should be an Object instance so Object.keys might get us what we need.
I'm not super sure so I'll have to experiment and check :D